### PR TITLE
Add identity deletion debug APIs

### PR DESCRIPTION
**Pull request by sebbi08**
# Readiness checklist

-   [ ] I added/updated tests.
-   [ ] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: src/modules/coreHttpApi/debug-controllers/IdentityDeletionController.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { Envelope } from "../../../infrastructure";
+import { BaseController } from "../common/BaseController";
+
+@Path("/api/v2/IdentityDeletion")
+export class IdentityDeletionController extends BaseController {
+    public constructor(@Inject private readonly transportServices: TransportServices) {
+        super();
+    }
+
+    @DELETE
```
Comment (jkoenig134) [inline src/modules/coreHttpApi/debug-controllers/IdentityDeletionController.ts:13]: I see what you did here - "deleting" the identity using the delete keyword, but this is not REST conform.

1. rename everything here from IdentityDeletion to IdentityDeletionProcesses
2. make this a POST
3. rename this function please, and be more cautious with copy pasting code

Additionally this feature is non-sense if we don't allow to cancel / query the identity deletion process, please add this as routes.

File: src/modules/coreHttpApi/debug-controllers/IdentityDeletionController.ts
Comment (jkoenig134) [inline src/modules/coreHttpApi/debug-controllers/IdentityDeletionController.ts:1]: this is not tested.

Comment (sebbi08) [reply src/modules/coreHttpApi/debug-controllers/IdentityDeletionController.ts:1]: should we test the debug API? 

Also do you want it to be added to the SDK or should we tests it without the SDK?

Comment (jkoenig134) [reply src/modules/coreHttpApi/debug-controllers/IdentityDeletionController.ts:1]: yes, test it, but only with axios

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**